File: app/src/main/java/zhan/rnadomgirdanimation/SimpleAdapter.java
package zhan.rnadomgirdanimation;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

import zhan.rnadomgirdanimation.refresh.IRefreshAdapter;

public class SimpleAdapter extends IRefreshAdapter<Integer> {

    @NotNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new SimpleHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item, parent, false));
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        ((SimpleHolder) holder).bind(mData.get(position));
    }
}

File: app/src/main/java/zhan/rnadomgirdanimation/SimpleAdapter.kt
package zhan.rnadomgirdanimation

import android.view.LayoutInflater
import android.view.ViewGroup

import androidx.recyclerview.widget.RecyclerView

import zhan.rnadomgirdanimation.refresh.IRefreshAdapter

class SimpleAdapter : IRefreshAdapter<Int>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return SimpleHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.item, parent, false))
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as SimpleHolder).bind(mData!![position])
    }
}
